Gruntfile.js

```diff
@@ -230,6 +237,30 @@ module.exports = function (grunt) {
 	grunt.registerTask("lint", ["tslint:build"]);
 	grunt.registerTask("all", ["clean", "test", "lint"]);
 	grunt.registerTask("rebuild", ["clean", "ts:devlib"]);
-	grunt.registerTask("default", "ts:devlib");
+	grunt.registerTask("default", ["ts:devlib", "generate_unit_testing_dependencies"]);
 	grunt.registerTask("docs-jekyll", ['template']);
 };
+
+function registerTestingDependenciesTasks(grunt) {
+	const configsBasePath = path.join(__dirname, "config");
+	const generatedVersionFileName = "test-deps-versions-generated.json";
+
+	grunt.registerTask("generate_unit_testing_dependencies", async function () {
+		var done = this.async();
+		const dependenciesVersions = {};
+		const testDependencies = grunt.file.readJSON(path.join(configsBasePath, "test-dependencies.json"));
+		for (var dependency of testDependencies) {
+			const dependencyVersion = await latestVersion(dependency.name);
+			dependenciesVersions[dependency.name] = dependencyVersion;
+		}
+		grunt.file.write(path.join(configsBasePath, generatedVersionFileName), JSON.stringify(dependenciesVersions));
+		done();
+	});
+
+	grunt.registerTask("verify_unit_testing_dependencies", function () {
+		if (!grunt.file.exists(path.join(configsBasePath, generatedVersionFileName))) {
```

Comment:
```
const generatedVersionFilePath = path.join(configsBasePath, generatedVersionFileName);
```
Can be used here and https://github.com/NativeScript/nativescript-cli/pull/4418/files#diff-35b4a816e0441e6a375cd925af50752cR256